snow/validators/validator.go

```diff
@@ -26,6 +30,49 @@ type Validator struct {
 // a validator of the Avalanche Network for the output of GetValidator.
 type GetValidatorOutput struct {
 	NodeID    ids.NodeID
-	PublicKey *bls.PublicKey
+	PublicKey *PublicKey
 	Weight    uint64
 }
+
+func NewPublicKey(key *bls.PublicKey) *PublicKey {
+	return &PublicKey{
+		key:   key,
+		bytes: nil,
+	}
+}
+
+func NewPublicKeyFromBytes(bytes []byte) *PublicKey {
+	return &PublicKey{
+		key:   nil,
+		bytes: bytes,
+	}
+}
+
+// PublicKey lazily serializes/deserializes the validator's key.
+// At least one of (key, bytes) must be provided.
+// This data structure is NOT thread-safe.
+type PublicKey struct {
+	key   *bls.PublicKey
+	bytes []byte
+}
+
+func (k *PublicKey) Key() (*bls.PublicKey, error) {
+	if k.key == nil {
+		pk := new(bls.PublicKey).Deserialize(k.bytes)
```

Comment:
nice :+1:  the one thing you will want to do here is ensure there is a note/warning with regards to the lack of bls sig verification. I believe this is totally fine in the context of gRPC as we can make assumptions that verification has already happened.

Otherwise someone might use the wrong implementation.
https://github.com/ava-labs/avalanchego/blob/ef6a2a2f7facd8fbefd5fb2ac9c4908c2bcae3e2/utils/crypto/bls/public.go#L33-L42

Reply:
Will add a note that we just assume that the bytes provided to us are correct